integrations/SnapPixel/util.js

```diff
@@ -158,6 +158,41 @@ const ecommEventPayload = (event, message) => {
       };
       break;
     }
+    case "product viewed":
+      payload = {
+        ...payload,
+        item_ids: get(message, "properties.product_id"),
+      };
+      break;
+    case "product list viewed": {
+      let itemIds = [];
+      const products = get(message, "properties.products");
+      if (isDefinedAndNotNull(products)) {
+        products.forEach((element, index) => {
+          const pId = element.product_id;
+          if (pId) {
+            itemIds.push(pId);
+          } else {
+            logger.debug(
+              `product_id not present for product at index ${index}`
+            );
+          }
+        });
+      } else {
+        itemIds = null;
```

Comment:
Why `null` and not an empty array?

Reply:
It's null because it gets eliminated at the end if it is empty.